node/coroutines/src/client_task.rs

```diff
@@ -228,25 +285,63 @@ impl ClientTask {
                 );
             }
         };
-
-        if let BlockProductionResult::Success(produced_beacon_block, produced_shard_block) =
-            self.client.try_produce_block(consensus_block_header.index, payload)
-        {
-            // Send block announcement.
+        let (mut beacon_block, mut shard_block, shard_extra) =
+            self.client.prepare_block(consensus_block_header.index, payload);
+        let (owner_uid, mapping) =
+            self.client.get_uid_to_authority_map(consensus_block_header.index);
+        if let Some(o) = owner_uid {
+            let beacon_signature = beacon_block.sign(self.client.signer.clone());
```

Comment:
I don't have good reasoning either way, but interested if you do - why not sign with self when `prepare_block`?